dot/parachain/dispute/overseer/signal.go

```diff
@@ -0,0 +1,46 @@
+package overseer
+
+import "github.com/ChainSafe/gossamer/lib/common"
+
+// Signal represents a signal sent from overseer
+type Signal[data any] struct {
+	Data            data
+	ResponseChannel chan<- any
+}
```

Comment:
a signal does not require a response. It is a signal, from overseer to subsystems and then subsystems would react/process based on that. Some subsystems, don't react at all.